cawg_identity/src/validator.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2025 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+//! This module contains the APIs you will use to validate a
+//! C2PA Manifest that contains one or more CAWG identity assertions.
+
+use async_trait::async_trait;
+use c2pa::{
+    dynamic_assertion::{PartialClaim, PostValidatorAsync},
+    ManifestAssertion,
+};
+use c2pa_status_tracker::StatusTracker;
+use serde_json::Value;
+
+use crate::IdentityAssertion;
+
+/// Validates a CAWG identity assertion.
+pub struct CawgValidator;
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+impl PostValidatorAsync for CawgValidator {
```

Comment:
Most of the _type_ names we've built in c2pa-rs and friends have `Async` as a prefix, not a suffix. Any reason for this one to be different?

(Note: Function names do have `_async` as a suffix.)

Reply:
I switched that around.